torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -121,6 +121,16 @@ c10::optional<at::Device> pickDeviceType(
   return device;
 }
 
+c10::optional<at::Device> pickDeviceType(const std::shared_ptr<Graph>& graph) {
```

Comment:
What if a graph has both cpu and cuda nodes? Should we error out?

Reply:
Yeah, fuser would probably take care of that, but I was thinking about the AOT scenario where we don't have the fuser pass safeguard :)